src/e2e/java/teammates/e2e/util/JMeterConfig.java

```diff
@@ -137,11 +148,58 @@ public HashTree createTestPlan() {
         // Create Test plan
         HashTree testPlanHashTree = new ListedHashTree();
         HashTree threadGroupHashTree = testPlanHashTree.add(testPlan, threadGroup);
+
+        // Add Config elements
         threadGroupHashTree.add(csvDataSet);
         threadGroupHashTree.add(cookieManager);
         threadGroupHashTree.add(defaultSampler);
-        threadGroupHashTree.add(onceOnlyController, loginSampler);
-        threadGroupHashTree.add(apiSampler);
+
+        // Add HTTP samplers (and configurations, pre/post processors)
+        HashTree loginSamplerHashTree = threadGroupHashTree.add(onceOnlyController, loginSampler);
+        HashTree apiSamplerHashTree = threadGroupHashTree.add(apiSampler);
+
+        switch (getTestMethod()) {
+        case HttpDelete.METHOD_NAME:
+            // fallthrough
+        case HttpPut.METHOD_NAME:
+            // fallthrough
+        case HttpPost.METHOD_NAME:
+            // Regex Extractor for CSRF token
+            RegexExtractor regexExtractor = new RegexExtractor();
+            regexExtractor.setName("Regular Expression Extractor");
+            regexExtractor.setUseField("true"); // Find matches in response headers
+            regexExtractor.setRefName("csrfToken");
+            regexExtractor.setRegex("Set-Cookie: CSRF-TOKEN=(.+?);");
```

Comment:
Is it safe to prepend `Set-Cookie`? If some other cookies are set and `CSRF-TOKEN` is not the first in the list, this matcher is not going to work.